transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -810,6 +822,10 @@ public final void flush() {
                 return;
             }
 
+            flush(outboundBuffer);
```

Comment:
> I think if a channel is configured to autoflush, it should just ignore requests from the application to flush

I'm not sure if this is a good thing todo. If the application wants to "force" a flush they may be doing so for good reason. I don't think we need to mix `ELIDE_FLUSHING` and `AUTO_FLUSH` as they seem like related but independent approaches. `ELIDE_FLUSHING` can be used by folks who explicitly flush to help optimize/combine flushes, and `AUTO_FLUSH` can be used by folks who don't want to bother calling flush at all (but still can explicitly flush if they have good reason to do so).

However this may require removing explicit flushes in handlers and instead relying on these "flushing strategies".